component/pyroscope/scrape/target.go

```diff
@@ -43,12 +43,16 @@ const (
 
 // Target refers to a singular HTTP or HTTPS endpoint.
 type Target struct {
+	// All labels of this target - public and private
+	allLabels labels.Labels
+	// Only public labels that are added to this target and its metrics.
+	publicLabels labels.Labels
```

Comment:
small nit: do we need to store a copy of the public labels?

AFAICT we only use it on the t.Labels() method. The method in turn is used a) In some logging contexts, b) when calling Clone() which will re-filter for the non-public labels, and c) in scrape_loop.go when checking whether to add targets in the `actives` slice, which should be already filtered inside of `populateLabels` in `targetsFromGroup`.

I think this may need a second look, but isn't a blocker in any case.

Reply:
+1 - it also seems to me we can get rid of `.Clone()` method - which would allow to remove even more stuff and simplify. Is there some magic that requires the `Target` to have `Clone()`?

The `t.Labels()` is [also used in here, btw, to determine which targets are active](https://github.com/grafana/agent/blob/main/component/pyroscope/scrape/scrape_loop.go#L70).

And speaking of logging, [this debug line will](https://github.com/grafana/agent/blob/main/component/pyroscope/scrape/scrape_loop.go#L274) always build the `Labels().String()` which adds to allocation. 

Maybe we should do those improvements in a follow-up PR too?

Reply:
Yeah I saw the `c)` use-case, but I _think_ that this will always be true since `targetsFromGroup` will always filter out targets that only have internal labels, and they'd be bundled into `dropped` instead

As I said, not a blocker from me, happy to see this land and see the benefits.